homeassistant/components/mysensors/device.py

```diff
@@ -19,33 +32,99 @@
+        self.node_id: int = node_id
+        self.child_id: int = child_id
+        self.value_type: int = value_type  # value_type as int. string variant can be looked up in gateway consts
+        self.child_type = self._mysensors_childsensor.type
         self._values = {}
         self._update_scheduled = False
         self.hass = None
 
+    @property
+    def dev_id(self) -> DevId:
+        """Return the DevId of this device.
+
+        It is used to route incoming MySensors messages to the correct device/entity.
+        """
+        return self.gateway_id, self.node_id, self.child_id, self.value_type
+
+    @property
+    def _logger(self):
+        return logging.getLogger(f"{__name__}.{self.name}")
+
+    async def async_will_remove_from_hass(self):
+        """Remove this entity from home assistant."""
+        for platform in PLATFORM_TYPES:
+            platform_str = MYSENSORS_PLATFORM_DEVICES.format(platform)
+            if platform_str in self.hass.data[DOMAIN]:
+                platform_dict = self.hass.data[DOMAIN][platform_str]
+                if self.dev_id in platform_dict:
+                    if platform_dict[self.dev_id] is not self:
+                        self._logger.warning(
+                            "possible duplicate device: %s", self.dev_id
+                        )
+                    del platform_dict[self.dev_id]
+                    self._logger.debug(
+                        "deleted %s from platform %s", self.dev_id, platform
+                    )
+
+    @property
+    def _mysensors_sensor(self) -> Sensor:
+        return self.gateway.sensors[self.node_id]
+
+    @property
+    def _mysensors_childsensor(self) -> ChildSensor:
+        return self._mysensors_sensor.children[self.child_id]
+
+    @property
+    def sketch_name(self) -> str:
+        """Return the name of the sketch running on the whole node (will be the same for several entities!)."""
+        return self._mysensors_sensor.sketch_name
+
+    @property
+    def sketch_version(self) -> str:
+        """Return the version of the sketch running on the whole node (will be the same for several entities!)."""
+        return self._mysensors_sensor.sketch_version
+
+    @property
+    def node_name(self) -> str:
+        """Name of the whole node (will be the same for several entities!)."""
+        return f"{self.sketch_name} {self.node_id}"
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID for use in home assistant."""
+        return f"mys{self.gateway_id}-{self.node_id}-{self.child_id}-{self.value_type}"
+
+    @property
+    def device_info(self) -> Optional[Dict[str, Any]]:
+        """Return a dict that allows home assistant to puzzle all entities belonging to a node together."""
+        return {
+            "identifiers": {(DOMAIN, f"mys{self.gateway_id}-{self.node_id}")},
+            "name": self.node_name,
+            "manufacturer": DOMAIN,
+            "model": self.node_name,
```

Comment:
If model is the same as name we don't need to add it. It's not a real model name.